src/ui/server/helpers/summary-lists/company-house-summary-list.test.ts

```diff
@@ -0,0 +1,173 @@
+import { format } from 'date-fns';
+import { generateFieldGroups, getKeyText, generateSummaryListRows, companyHouseSummaryList, generateAddressHTML } from './company-house-summary-list';
+import { SummaryListItemData } from '../../../types';
+import { FIELD_IDS } from '../../constants';
+import { FIELDS, PAGES } from '../../content-strings';
+import { mockCompanyResponse } from '../../test-mocks';
+
+const {
+  EXPORTER_BUSINESS: { COMPANY_HOUSE },
+} = FIELD_IDS.INSURANCE;
+
+const { COMPANY_NAME, COMPANY_ADDRESS, COMPANY_NUMBER, COMPANY_INCORPORATED, COMPANY_SIC } = COMPANY_HOUSE;
+
+describe('server/helpers/summary-lists/company-house-summary-list', () => {
+  describe('generateAddressHTML()', () => {
+    it('should generate address where all fields are present', () => {
+      const address = {
+        careOf: 'Careof',
+        premises: 'Premise',
+        addressLine1: 'Line 1',
+        addressLine2: 'Line 2',
+        locality: 'Locality',
+        region: 'Region',
+        postalCode: 'Postcode',
+        country: 'Country',
+        __typename: 'CompanyAddress',
+      };
+
+      const result = generateAddressHTML(address);
+
+      const expected = `${address.careOf}<br>${address.premises}<br>${address.addressLine1}<br>${address.addressLine2}<br>${address.locality}<br>${address.region}<br>${address.postalCode}<br>${address.country}<br>`;
+      expect(result).toEqual(expected);
+    });
+
+    it('should only generate address where all fields are not null', () => {
+      const address = {
+        careOf: null,
+        premises: 'Premise',
+        addressLine1: 'Line 1',
+        addressLine2: null,
+        locality: 'Locality',
+        region: 'Region',
+        postalCode: 'Postcode',
+        country: null,
+        __typename: 'CompanyAddress',
```

Comment:
I don't think we need `__typename` in these kinds of tests(?), same as above

Reply:
Have kept typename as it is in the address when going into the function and want to make sure that it is not printed on the page